boost-maven/boost-maven-plugin/src/main/java/io/openliberty/boost/docker/dockerizer/DockerizeSpringBootClasspath.java

```diff
@@ -0,0 +1,240 @@
+                        fos.write(buffer, 0, len);
+                    }
+                    fos.close();
+                }
+                ze = zis.getNextEntry();
+            }
+            // close last ZipEntry
+            zis.closeEntry();
+            zis.close();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+    }
+
+    private String getSpringStartClass(File artifact) {
+        try (JarFile jarFile = new JarFile(artifact)) {
+            Manifest manifest = jarFile.getManifest();
+            if (manifest != null) {
+                Attributes attributes = manifest.getMainAttributes();
+                return attributes.getValue("Start-Class");
+            }
+        } catch (IOException e) {}
+        return "";
+    }
+
+    private File createNewDockerFile(File dockerfileDirectory) throws IOException {
+        File dockerFile = new File(dockerfileDirectory, "Dockerfile");
+        Files.createFile(dockerFile.toPath());
+        log.info("Creating Dockerfile: " + dockerFile.getAbsolutePath());
+        return dockerFile;
+    }
+
+    private void writeSpringBootDockerFile(File dockerFile, String startClass) throws IOException {
+        ArrayList<String> lines = new ArrayList<>();
+        lines.add(BOOST_GEN);
+        lines.add("FROM adoptopenjdk/openjdk8-openj9");
+        lines.add("VOLUME /tmp");
+        lines.add("ARG DEPENDENCY=target/dependency");
+        lines.add("COPY ${DEPENDENCY}/BOOT-INF/lib /app/lib");
+        lines.add("COPY ${DEPENDENCY}/META-INF /app/META-INF");
+        lines.add("COPY ${DEPENDENCY}/BOOT-INF/classes /app");
+        lines.add("ENTRYPOINT [\"java\",\"-cp\",\"app:app/lib/*\",\"" + startClass + "\"]");
+        Files.write(dockerFile.toPath(), lines, Charset.forName("UTF-8"));
+    }
+
+    /**
+     * Creates a new .dockerignore file in the project directory if the
+     * .dockerignore file doesn't exist. Existing .dockerignore file will be
+     * appended with the lines generated by the plugin. Creating/appending of
+     * .dockerignore happens only when the plugin generates Dockerfile.
+     *
+     * @throws Exception
+     */
+    public void createDockerIgnore() throws Exception {
+        File dockerIgnore = new File(projectDirectory, ".dockerignore");
+        boolean dockerIgnoreAlreadyExists = false;
+
+        if (isFileGeneratedByPlugin(new File(projectDirectory, "Dockerfile"))) {
+            try {
+                Files.createFile(dockerIgnore.toPath());
+                log.info("Creating .dockerignore: " + dockerIgnore.getAbsolutePath());
+            } catch (FileAlreadyExistsException e) {
+                log.warn("The .dockerignore file already exists");
+                dockerIgnoreAlreadyExists = true;
+            }
+            writeDockerIgnore(dockerIgnore, dockerIgnoreAlreadyExists);
+        }
+    }
+
+    private void writeDockerIgnore(File dockerIgnore, boolean dockerIgnoreAlreadyExists) throws IOException {
```

Comment:
Won't we have a problem here if there are multiple invocations of docker-build with our generated Dockerfile? Seems like it would just keep appending forever.

Reply:
Never mind, saw that you check `isFileGeneratedByPlugin` for the dockerignore as well.